storage/engine/rocksdb/db.cc

```diff
@@ -1105,32 +1105,6 @@ class BaseDeltaIterator : public rocksdb::Iterator {
                                base_iterator_->key());
   }
 
-  void AssertInvariants() {
-#ifndef NDEBUG
-    if (!Valid()) {
-      return;
-    }
-    if (!BaseValid()) {
-      assert(!current_at_base_ && delta_iterator_->Valid());
-      return;
-    }
-    if (!DeltaValid()) {
-      assert(current_at_base_ && base_iterator_->Valid());
-      return;
-    }
-    // we don't support those yet
-    assert(delta_iterator_->Entry().type != rocksdb::kLogDataRecord);
-    int compare = kComparator.Compare(delta_iterator_->Entry().key,
-                                      base_iterator_->key());
-    // current_at_base -> compare < 0
-    assert(!current_at_base_ || compare < 0);
-    // !current_at_base -> compare <= 0
-    assert(current_at_base_ && compare >= 0);
-    // equal_keys_ <=> compare == 0
-    assert((equal_keys_ || compare != 0) && (!equal_keys_ || compare == 0));
-#endif
-  }
-
   void Advance() {
```

Comment:
Could you add comments on all of the methods here? CheckPrefix, ProcessDelta, ... - I was reading all of this code and it wasn't fun to try to guess what it wants to do. Ditto for the base and delta iterator and the other miscellaneous state on this object.